Implement the SonarCloud stage

bellingard opened the pull request:
@rombert At line #69, you need to update the code to retrieve your SonarCloud user token (that you made available just for the 'Sling' folder on Jenkins) so that it is passed instead of the fake "SC_TOKEN" variable that I've used.

bellingard committed:
Implement the SonarCloud stage

rombert commented on vars/slingOsgiBundleBuild.groovy:
```diff
@@ -63,7 +63,20 @@ def call(Map params = [:]) {
+                                sonarcloudParams="${sonarcloudParams} -Dsonar.branch.name=${BRANCH_NAME}"
+                            }
+                            // Alls params are set, let' execute
+                            def mvnCmd = "mvn -U clean verify sonar:sonar ${sonarcloudParams}"
```
Thanks for the PR @bellingard ! I'll add the token to the execution, but was wondering whether we need to use the trick from https://github.com/apache/sling-tooling-jenkins/blob/eb00228b2e85132058adde6561d00afea20cc3b2/vars/slingOsgiBundleBuild.groovy#L51-L53 to prevent the `sonar.login` token from being printed in the Jenkins build log?

bellingard replied:
I've improved the code to wrap the command using withCredentials, this will mask the token if it is printed out.
Also, note that I've used a "secret text" type credential (named "robert_token" in this example). This is what you should set on the Sling folder in Jenkins.

rombert replied:
Thanks @bellingard . I'll merge this as-is and do any necessary tweaks later.

bellingard committed:
Wrap the Maven execute using withCredentials

withCredentials() will set the SONAR_TOKEN variable which is recognized by the scanner.
It will also automatically mask the token if it is ever printed out in the logs.

rombert commented:
Merged, thanks @bellingard
